src/Microsoft.DocAsCode.Build.Engine/PostProcessors/SitemapGenerator.cs

```diff
@@ -0,0 +1,167 @@
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.IO;
+    using System.Linq;
+    using System.Xml.Linq;
+
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Glob;
+    using Microsoft.DocAsCode.Plugins;
+
+    // TODO: support incremental and update lastmod only when src file is changed, blocked by the postprocessor incremental framework
+    [Export(nameof(SitemapGenerator), typeof(IPostProcessor))]
+    public class SitemapGenerator : IPostProcessor
+    {
+        private static readonly string NamespaceName = "http://www.sitemaps.org/schemas/sitemap/0.9";
+        private const string HtmlExtension = ".html";
+        private const string SitemapName = "sitemap.xml";
+
+        public string Name => nameof(SitemapGenerator);
+
+        public ImmutableDictionary<string, object> PrepareMetadata(ImmutableDictionary<string, object> metadata)
+        {
+            return metadata;
+        }
+
+        public Manifest Process(Manifest manifest, string outputFolder)
+        {
+            if (string.IsNullOrEmpty(manifest.SitemapOptions?.BaseUrl))
+            {
+                return manifest;
+            }
+
+            if (!manifest.SitemapOptions.BaseUrl.EndsWith("/"))
+            {
+                manifest.SitemapOptions.BaseUrl += '/';
+            }
+
+            if (!Uri.TryCreate(manifest.SitemapOptions.BaseUrl, UriKind.Absolute, out var baseUri))
+            {
+                Logger.LogWarning($"Base url {manifest.SitemapOptions.BaseUrl} is not in a valid uri format.");
+                return manifest;
+            }
+
+            if (manifest.SitemapOptions.Priority.HasValue && (manifest.SitemapOptions.Priority < 0 || manifest.SitemapOptions.Priority > 1))
+            {
+                Logger.LogWarning($"Invalid priority {manifest.SitemapOptions.Priority}, priority must be between 0.0 and 1.0. Use default value 0.5 instead");
+                manifest.SitemapOptions.Priority = 0.5;
+            }
+
+            XElement urlset = new XElement(XName.Get("urlset", NamespaceName));
+            XDocument sitemapDocument = new XDocument(urlset);
+
+            foreach (var file in GetHtmlOutputFiles(manifest).OrderBy(s => s.Item1))
+            {
+                var options = GetOptions(manifest.SitemapOptions, file.Item1);
+
+                var currentBaseUri = baseUri;
+                if (options.BaseUrl != manifest.SitemapOptions.BaseUrl)
+                {
+                    if (!Uri.TryCreate(options.BaseUrl, UriKind.Absolute, out currentBaseUri))
+                    {
+                        Logger.LogWarning($"Base url {options.BaseUrl} is not in a valid uri format, use base url from the default setting {manifest.SitemapOptions.BaseUrl} instead.");
+                        currentBaseUri = baseUri;
+                    }
+                }
+
+                urlset.Add(GetElement(file.Item2.RelativePath, currentBaseUri, options));
```

Comment:
considering: [XStreamingElement](https://docs.microsoft.com/en-us/dotnet/api/system.xml.linq.xstreamingelement?view=netframework-4.7) + [Linq](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-perform-streaming-transform-of-large-xml-documents)